### DISCUSSION TITLE
How to associate `winget` with a Microsoft Store account?

### DISCUSSION THREAD

**Discussion by jpluimers** [microsoft/winget-cli#1992] [Q&A]
In https://github.com/microsoft/winget-cli/issues/1585 it is clear that (at least for the foreseeable future) you need a Microsoft Store account to install applications with the `msstore` source.

I tried finding out how to have `winget` associate itself with such an account, but could not find it in the command-line options, or documented.

I tried all these to no avail:

- `winget source update msstore`
- `winget settings` followed by reading through <https://aka.ms/winget-settings> (which redirects to <https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md>) which does not mention "store" at all
- `winget features`
- searching with:
  - <https://docs.microsoft.com/en-us/search/?terms=msstore>
  - <https://www.google.com/search?q=%22msstore%22+site%3Ahttps%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fwindows%2Fpackage-manager%2Fwinget>
  - <https://www.google.com/search?q=%22Microsoft+Store%22+site%3Ahttps%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fwindows%2Fpackage-manager%2Fwinget>

So: how do I make `winget` to associate itself with my Microsoft Store account for the `msstore` source, preferable from the CLI?

**Comment by ItzLevvie** [comment]


**Reply by jpluimers** [reply]
Thanks @ItzLevvie.

I'm bad at abbreviations, but guessing AAD has to do with Active Directory on a company domain, and the Microsoft Store Account is only available with Windows 11: _is there a Windows 10 way to do it without such an Active Directory account_?

**Reply by jpluimers** [reply]
No problem and thanks, it works!

Full steps to logon with a Microsoft account on the Microsoft Store app of your Windows 10 machine:

1. Either of these:
   - From the CLI run `start ms-windows-store:`, or 
   - from the task bar press `Win`+`R` followed by typing `ms-windows-store:` and the `Enter` key, or 
   - on the task bar search for `Store`, then click `Microsoft Store`:
     ![image](https://user-images.githubusercontent.com/2033367/156838979-4b7be36a-e1aa-48c2-80ba-73ae354b4c37.png)
2. In the title bar, click on the profile icon (it looks like a very abstract person icon, similar to "ጰ"):
   ![image](https://user-images.githubusercontent.com/2033367/156841025-c6ec619a-40cf-4346-a82c-0e9030dab3d1.png)
3. From the pop-up menu, choose "Sign in"
   ![image](https://user-images.githubusercontent.com/2033367/156841379-a6be4656-dbaf-440c-84cb-5ec2502aec53.png)
4. The sign on dialog appears: 
   ![image](https://user-images.githubusercontent.com/2033367/156841506-38a92ba0-c063-4030-84d9-275400e9c2bd.png)
5. Enter credentials, for instance begin with email, then press "Next":
   ![image](https://user-images.githubusercontent.com/2033367/156848049-e32d590f-f616-4422-bcff-b3fe6b098fb1.png)
6. then password, then press "Sign in":
   ![image](https://user-images.githubusercontent.com/2033367/156848729-ec36ef3b-711e-448b-a0a4-f94703557f58.png)
7. finally choose if you want to use this account for everything on your Windows installation ("Next" button) or just the Microsoft Apps ("Microsoft apps only" hyperlink)
   ![image](https://user-images.githubusercontent.com/2033367/156875546-a3b39466-d596-46e1-9155-7f1ab372e944.png)
8. If the abstract person icon has changed into your account initials (in my case "WM"), then you are set:
    ![image](https://user-images.githubusercontent.com/2033367/156849050-da8c33b4-6a7f-45e2-a6c1-0f441cccae0a.png)

Back on the CLI you can now install an application that in the Windows Store does not use an account, but winget requires to have one, in my case it was Microsoft To Do:

```
winget install --accept-source-agreements --accept-package-agreements 9NBLGGH5R558
````

Full log of that:

``` text
C:\temp> winget install --accept-source-agreements --accept-package-agreements 9NBLGGH5R558
Found Microsoft To Do: Lists, Tasks & Reminders [9NBLGGH5R558] Version Unknown
This package is provided through Microsoft Store. winget may need to acquire the package from Microsoft Store on behalf of the current user.
Version: Unknown
Publisher: Microsoft Corporation
Publisher Url: https://go.microsoft.com/fwlink/?linkid=846683
Publisher Support Url: https://go.microsoft.com/fwlink/?linkid=846682
Description: Got something on your mind? Get Microsoft To Do. Whether you want to increase your productivity, decrease your stress levels, or just free up some mental space, Microsoft To Do makes it easy to plan your day and manage your life.

With Microsoft To Do, you can:
• Stay focused with My Day, a personalized daily planner with suggested tasks
• Get your lists anywhere, on any device
• Share lists and assign tasks with your friends, family, colleagues, and classmates
• Personalize your lists with bold and colorful backgrounds
• Set one-time or recurring due dates and reminders
• Break your tasks into manageable steps
• Add notes to any task
• Attach files up to 25 MB to any task
• Sync your tasks between Outlook and To Do
• Group your lists together by topic or project
Whether it's for work, school, or home, To Do helps you organize and simplify your plans. To Do is free and available on all your devices.

Learn more: https://to-do.microsoft.com
Follow us on Twitter: @MicrosoftToDo
Need support? https://microsoft-todo.helpshift.com
By installing Microsoft To Do, you agree to the Microsoft Terms of Use: https://go.microsoft.com/fwlink/?linkid=842575
License: https://go.microsoft.com/fwlink/?linkid=842576
Privacy Url: https://privacy.microsoft.com/privacystatement
Copyright: © Microsoft Corporation
Agreements:
Category: Productivity
Pricing: Free
Free Trial: No
Terms of Transaction: https://aka.ms/microsoft-store-terms-of-transaction
Seizure Warning: https://aka.ms/microsoft-store-seizure-warning
Store License Terms: https://aka.ms/microsoft-store-license


Verifying/Requesting package acquisition...
Verifying/Requesting package acquisition success
Starting package install...
  ██████████████████████████████  100%
Successfully installed
C:\temp>
```

--jeroen
